src/express/commands/monitor.js

```diff
@@ -62,15 +108,29 @@ export async function monitor() {
             console.log("📍Awaiting Checkpoint 🚌")
         }
 
-        let stateSyncTx = await checkStateSyncTx(machine0);
-        if (stateSyncTx) {
-            console.log("📍StateSync found ✅ ; Tx_Hash: ", stateSyncTx);
+        
+        var firstStateSyncTx = await checkStateSyncTx(machine0,1);
+        if (firstStateSyncTx) {
+            let timeOfFirstStateSyncTx = firstStateSyncTx.record_time
+            let firstEpochTime = parseInt(new Date(timeOfFirstStateSyncTx).getTime() / 1000);
+            let currentEpochTime = parseInt(new Date().getTime() / 1000);
+            let stateSyncTxList = await getStateSyncTxList(machine0,firstEpochTime,currentEpochTime);
+            if (stateSyncTxList) {
+                
+                let lastStateID =  stateSyncTxList.length
+                let lastStateSyncTxHash = stateSyncTxList[lastStateID-1].tx_hash
+                console.log("📍StateSyncs found on Heimdall ✅ ; Count: ", lastStateID, " ; Last Tx Hash: ", lastStateSyncTxHash);
+            }
+            
         } else {
             console.log("📍Awaiting StateSync 🚌")
         }
 
-        if (checkpointCount > 0 && stateSyncTx) {
-            break;
+        let lastStateId = await lastStateIdFromBor(machine0);
```

Comment:
i hope that we are doing this after we are sure that that state sync has reached bor
otherwise if the state sync is in heimdall and hasn't reached on bor and we are making this call then there might be incorrect result
let me know if i am missing something

Reply:
It will return 0 incase bor did not receive any statesync. Also, we have two places where we print last state ID. One for heimdall and one for bor.
L122 and L131.